Presentation.Web/app/components/data-processing/tabs/data-processing-agreement-tab-roles.controller.ts

```diff
@@ -0,0 +1,271 @@
+﻿module Kitos.DataProcessing.Agreement.Edit.Roles {
+    "use strict";
+
+    export class EditRolesDataProcessingAgreementController {
+        static $inject: Array<string> = [
+            "dataProcessingAgreementService",
+            "notify",
+            "hasWriteAccess",
+            "dataProcessingAgreement",
+            "dataProcessingAgreementRoles",
+            "select2LoadingService"
+        ];
+
+        constructor(
+            private readonly dataProcessingAgreementService: Services.DataProcessing.IDataProcessingAgreementService,
+            private readonly notify,
+            public hasWriteAccess,
+            private readonly dataProcessingAgreement: Models.DataProcessing.IDataProcessingAgreementDTO,
+            private readonly dataProcessingAgreementRoles: Models.DataProcessing.IDataProcessingRoleDTO[],
+            private readonly select2LoadingService: Services.ISelect2LoadingService) {
+        }
+
+        headerName = this.dataProcessingAgreement.name;
+
+        roles = _.map(this.dataProcessingAgreementRoles, role => this.mapToRoleOptions(role));
+
+        selectedRoleIdAsString = this.roles[0].id.toString();
```

Comment:
Hvis der ikke er nogle roller (er muligt hvis lokal admin ikke har oprettet dem endnu) så fejler viewet.. det mangler du at tage højde for.